.github/workflows/phpunit.yml

```diff
@@ -20,29 +20,35 @@ jobs:
     strategy:
       matrix:
         drupal-version:
-          - '9'
-          - '10'
+          - '10.1'
         php-version:
-          - '8.1'
+          - '8.2'
         experimental: [false]
         do-leniency: [true]
+        patches: ["{}"]
         include:
-          # XXX: 7.4 To be removed when we are no longer actually running it anywhere.
-          - drupal-version: '9'
-            php-version: '7.4'
-            experimental: true
-            do-leniency: false
-          # XXX: D9 doesn't seem to be installable with PHP 8.2, due to
-          # incompatible dependencies.
-          - drupal-version: '9'
+          - drupal-version: '10.2'
             php-version: '8.2'
             experimental: true
             do-leniency: true
-          - drupal-version: '10'
-            php-version: '8.2'
+            patches: |-
+              {
+                "drupal/core": {
+                  "Referencable unpublished nodes": "https://www.drupal.org/files/issues/2024-01-07/2845144-78.patch"
+                }
+              }
+          - drupal-version: '10.2'
+            php-version: '8.3'
             # XXX: experimental, for now.
             experimental: true
             do-leniency: true
+            patches: |-
+              {
+                "drupal/core": {
+                  "Referencable unpublished nodes": "https://www.drupal.org/files/issues/2024-01-07/2845144-78.patch"
+                }
+              }
```

Comment:
Found that https://github.com/discoverygarden/phpunit-action/pull/19/files#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6R172-R174 was failing on D10.2, so slapped together a mechanism so different versions can pass their own patches... is kinda gross, but whatever.